runtime/compiler/optimizer/SPMDParallelizer.cpp

```diff
@@ -1441,6 +1441,30 @@ bool TR_SPMDKernelParallelizer::processSPMDKernelLoopForSIMDize(TR::Compilation
             {
             // increment of PIV
             traceMsg(comp, "Reducing the number of iterations of the loop %d at storeNode [%p] by vector length %d \n",loop->getNumber(), storeNode, unrollCount);
+	    TR_ASSERT_FATAL(storeNode->getFirstChild()->getOpCode().isAdd() || storeNode->getFirstChild()->getOpCode().isSub(), "PIV increment should be simple (either by add or by sub");
+	    TR_ASSERT_FATAL(storeNode->getFirstChild()->getFirstChild()->getOpCode().isLoad(), "PIV increment should have load");
+	    TR_ASSERT_FATAL(!storeNode->getFirstChild()->getSecondChild()->getSymbolReference(), "PIV increment should have const increment value");
```

Comment:
I think this assert was supposed to check if the child is a const.

Reply:
Code Updated